src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -165,46 +128,40 @@ public void should_record_generic_failure_when_metadata_parsing_fails() throws E
         processor.processBlobs();
 
         // then
-        List<Envelope> envelopesInDb = envelopeRepository.findAll();
-        assertThat(envelopesInDb).isEmpty();
-
-        // and
-        List<ProcessEvent> processEvents = processEventRepository.findAll();
-        assertThat(processEvents).hasSize(1);
-
-        ProcessEvent processEvent = processEvents.get(0);
-
-        assertThat(processEvent.getContainer()).isEqualTo(testContainer.getName());
-        assertThat(processEvent.getZipFileName()).isEqualTo(invalidMetafileZip);
-        assertThat(processEvent.getEvent()).isEqualTo(DOC_FAILURE);
-        assertThat(processEvent.getId()).isNotNull();
-        assertThat(processEvent.getReason()).isNotBlank();
+        envelopeWasNotCreated();
+        checkFailureEvent(invalidMetafileZip, DOC_FAILURE);
     }
 
     @Test
     public void should_record_generic_failure_when_zip_contains_documents_not_in_pdf_format() throws Exception {
+        // given
+        uploadZipToBlobStore("5_24-06-2018-00-00-00.zip");
+
+        // when
+        processor.processBlobs();
+
+        // then
+        envelopeWasNotCreated();
         checkFailureEvent("5_24-06-2018-00-00-00.zip", Event.DOC_FAILURE);
     }
 
     @Test
     public void should_record_signature_failure_when_zip_contains_invalid_signature() throws Exception {
+        // given
         processor.signatureAlg = "sha256withrsa";
         processor.publicKeyBase64 = getXyzPublicKey64();
-        checkFailureEvent("43_24-06-2018-00-00-00.test.zip", Event.DOC_SIGNATURE_FAILURE);
-    }
 
-    public void checkFailureEvent(String invalidZipFile, Event event) throws Exception {
-        // given
-        uploadZipToBlobStore(invalidZipFile);
+        uploadZipToBlobStore("43_24-06-2018-00-00-00.test.zip");
 
         // when
         processor.processBlobs();
 
         // then
-        List<Envelope> envelopesInDb = envelopeRepository.findAll();
-        assertThat(envelopesInDb).isEmpty();
+        envelopeWasNotCreated();
+        checkFailureEvent("43_24-06-2018-00-00-00.test.zip", Event.DOC_SIGNATURE_FAILURE);
+    }
 
-        // and
+    private void checkFailureEvent(String invalidZipFile, Event event) {
```

Comment:
it would be nice if we could rename variable invalidZipFile to just zipFile as in some case we are passing valid zip file for e.g VALID_ZIP_FILE_WITH_CASE_NUMBER

Reply:
fixed. also renamed method so that assertion reads better.